apps/prairielearn/elements/pl-multiple-choice/pl-multiple-choice.py

```
@@ -677,14 +745,32 @@ def grade_multiple_choice(submitted_key: str) -> tuple[float, str | None]:
+    builtin_grading = pl.get_boolean_attrib(
+        element, "builtin-grading", BUILTIN_GRADING_DEFAULT
+    )
+
+    number_answers = len(data["params"][name])
+    all_keys = list(it.islice(pl.iter_keys(), number_answers))
+
+    if not builtin_grading:
+        # Still test that valid and invalid submissions are handled correctly
+        result = data["test_type"]
+        if result in {"correct", "incorrect"}:
+            data["raw_submitted_answers"][name] = random.choice(all_keys)
```

Ruff (0.15.12)

[error] Standard pseudo-random generators are not suitable for cryptographic purposes

(S311)